targets/TARGET_STM/i2c_api.c

```diff
@@ -61,24 +59,40 @@
 #define I2C_NUM (5)
 static I2C_HandleTypeDef* i2c_handles[I2C_NUM];
 
-/* Timeout values are based on core clock and I2C clock.
-   The BYTE_TIMEOUT is computed as twice the number of cycles it would
+/* Timeout value based on Core and I2C clocks */
+#define BYTE_TIMEOUT (SystemCoreClock / obj_s->hz)
+
+/* The BYTE_TIMEOUT_20 is computed as twice the number of cycles it would
    take to send 10 bits over I2C. Most Flags should take less than that.
    This is for immediate FLAG or ACK check.
 */
-#define BYTE_TIMEOUT ((SystemCoreClock / obj_s->hz) * 2 * 10)
-/* Timeout values based on I2C clock.
-   The BYTE_TIMEOUT_US is computed as 3x the time in us it would
+#define BYTE_TIMEOUT_20 (BYTE_TIMEOUT * 2 * 10)
+
+/* The BYTE_TIMEOUT_30 is computed as 3x the time in us it would
    take to send 10 bits over I2C. Most Flags should take less than that.
    This is for complete transfers check.
 */
-#define BYTE_TIMEOUT_US   ((SystemCoreClock / obj_s->hz) * 3 * 10)
+#define BYTE_TIMEOUT_30 (BYTE_TIMEOUT * 3 * 10)
+
 /* Timeout values for flags and events waiting loops. These timeouts are
    not based on accurate values, they just guarantee that the application will
    not remain stuck if the I2C communication is corrupted. 
 */
 #define FLAG_TIMEOUT ((int)0x1000)
 
+/* This function assumes the tick is 1 us */
```

Comment:
At a minimum should also comment that it assumes it's 32-bit - not sure if@LMESTM is happy with that assumption for this patch, given his comments.